src/app/api/events/[event]/participants/register-participants/route.ts

```diff
@@ -0,0 +1,49 @@
+import { prisma } from "@/prisma"
+import { NextRequest, NextResponse } from "next/server"
+
+interface RegisterParticipantParams {
+  params: {
+    event: string
+  }
+}
+
+export async function POST(
+  request: NextRequest,
+  { params }: RegisterParticipantParams,
+) {
+  const { event } = params
+
+  const { name, email, github } = await request.json()
+
+  const existParticipant = await prisma.user.findFirst({
```

Comment:
I should put the `email` as an unique field. Forgot about that. I'll fix it, then you don't need to check try to find the user before creating.